lib/usd/translators/shading/mtlxBaseWriter.cpp

```diff
@@ -460,6 +499,24 @@ UsdAttribute MtlxUsd_BaseWriter::AddNormalMapping(UsdAttribute normalInput)
     return mapInput;
 }
 
+UsdAttribute MtlxUsd_BaseWriter::PreserveNodegraphBoundaries(UsdAttribute input)
+{
+    UsdShadeNodeGraph nodeGraph(GetNodeGraph());
```

Comment:
Top-level shader nodes like lambert and phong require connecting to ancillary nodes via a node graph boundary port. This check and adds such a port if required.